compendium/DeclarativeServices/src/manager/SingletonComponentConfiguration.cpp

```diff
@@ -152,7 +156,11 @@ void SingletonComponentConfigurationImpl::UnbindReference(const std::string& ref
                      std::current_exception());
   }
   auto context = GetComponentContext();
-  context->RemoveFromBoundServicesCache(refName, ref);
+  if (!context->RemoveFromBoundServicesCache(refName, ref)) {
+      GetLogger()->Log(cppmicroservices::logservice::SeverityLevel::LOG_ERROR,
+          "Failure when removing a reference from the BoundServices Cache");
+  }
```

Comment:
Perhaps I'll add a "return" here as well, in case any code is added after the if-statement in the future.